test/yargs-parser.js

```diff
@@ -176,6 +176,20 @@ describe('yargs-parser', function () {
     x.should.be.a('string').and.equal('56')
   })
 
+  it('should default numbers to undefined', function () {
+    var n = parser([ '-n' ], {
+      number: ['n']
+    }).n
+    expect(n).to.equal(undefined)
```

Comment:
@nexdrew what are your thoughts about defaulting a number to `undefined` ... but one could also argue for `NaN` or `0`.